tcms/testplans/static/testplans/js/get.js

```diff
@@ -157,15 +168,11 @@ function attachEvents(testCases, testPlanId, permissions) {
         $('.js-test-case-menu-status').click(function(ev) {
             const testCaseId = getCaseIdFromEvent(ev);
 
-            jsonRPC('TestCase.update', [testCaseId, {'case_status': ev.target.dataset.id}], function(tc) {
-                //todo: refactor when testcase_status is replaced with boolean flag
-                if (tc.case_status_id !== 2) {
-                    delete allTestCases.confirmed[testCaseId];
-                    allTestCases.not_confirmed[testCaseId] = tc;
-                    $(ev.target).closest(`[data-testcase-pk=${testCaseId}]`).fadeOut(fadeAnimationTime, function() {
-                        $(this).remove();
-                    });
-                }
+            jsonRPC('TestCase.update', [testCaseId, {'case_status': ev.target.dataset.id}], function() {
+                // todo show the case as not confirmed
```

Comment:
this will easily be resolved when you simplify the visualisation above and make it a function and just call it unconditionally.